### PR TITLE
assorted minor code refactoring in ParseXS

### PR DESCRIPTION
**Pull request by iabyn**
Various commits that rename lexical vars, or reduce their scope etc. Should be no functional changes.

### CODE REVIEW

File: dist/ExtUtils-ParseXS/lib/ExtUtils/ParseXS.pm [hunk 1]
```diff
@@ -1741,7 +1741,7 @@ EOF
   #      XSINTERFACE_FUNC_SET(cv, $value);
 
   print Q(<<"EOF") if defined $self->{XsubAliases} or defined $self->{interfaces};
-#    {
+#    [[
```
Comment (jkeenan) [inline dist/ExtUtils-ParseXS/lib/ExtUtils/ParseXS.pm:1767]: I'm not familiar enough with C to understand why you are suggesting replacing single braces with double brackets.  Can you explain?

**Review by jkeenan — COMMENTED** [review]
I'm working through this p.r. one commit at a time.

File: dist/ExtUtils-ParseXS/lib/ExtUtils/ParseXS.pm [hunk 2]
```diff
@@ -839,11 +848,14 @@ EOM
...
-    my $extra_args = 0;
-    my @args_num = ();
     my $num_args = 0;
     my $report_args = ''; # the arg's description as used by croak()
     my $ellipsis;
+    my $min_args;
+
+    {
+    my $extra_args = 0;
+    my @args_num = ();
```
Comment (jkeenan) [inline dist/ExtUtils-ParseXS/lib/ExtUtils/ParseXS.pm]: I believe that everything between lines 856 and 908 (exclusive) should be indented 2(?) spaces -- unless that's handled in a later commit.

**Review by jkeenan — COMMENTED** [review]
See comment inline.

**Comment by jkeenan** [comment]
Refactoring ExtUtils-ParseXS was the first work I did on the core back in 2010.  That's enough for one lifetime!  Thanks for taking this on.

File: dist/ExtUtils-ParseXS/lib/ExtUtils/ParseXS.pm (hunk 1 above)
Comment (iabyn) [reply dist/ExtUtils-ParseXS/lib/ExtUtils/ParseXS.pm:1767]: That's an escaped string. The 'Q' function converts [[ to { and ]] to }. So in the final emitted C code it is still a single brace, but in the perl source the unbalanced brace is hidden from text editors. This is already done many times in ParseXS.pm; this commit just changes the few places where it's not done.

File: dist/ExtUtils-ParseXS/lib/ExtUtils/ParseXS.pm (hunk 2 above)
Comment (iabyn) [reply dist/ExtUtils-ParseXS/lib/ExtUtils/ParseXS.pm]: The re-indentation is done in the follow-up commit.

**Review by jkeenan — APPROVED** [review]

**Pull request merged**